mirgecom/diffusion.py

```diff
@@ -337,6 +338,59 @@ def get_diffusion_flux(
             penalty_amount=penalty_amount)
 
 
+class PrescribedFluxDiffusionBoundary(DiffusionBoundary):
+    r"""Prescribed flux boundary condition for the diffusion operator.
+
+    For the boundary condition $(\nabla u \cdot \mathbf{\hat{n}})|_\Gamma$, uses
+    external data
+
+    .. math::
+
+        u^+ = u^-
+
+    when computing the boundary fluxes for $\nabla u$, and applies directly
+    the prescribed flux $g$ when computing $\nabla \cdot (\kappa \nabla u)$.
+
+    .. automethod:: __init__
+    .. automethod:: get_grad_flux
+    .. automethod:: get_diffusion_flux
+    """
+
+    def __init__(self, value):
+        """
+        Initialize the boundary condition.
+
+        Parameters
+        ----------
+        value: float or meshmode.dof_array.DOFArray
+            the value(s) of $g$ along the boundary
+        """
+        self.value = value
+
+    def get_grad_flux(self, dcoll, dd_bdry, kappa_minus, u_minus, *,
+            numerical_flux_func=grad_facial_flux_weighted):  # noqa: D102
+        actx = u_minus.array_context
+        kappa_tpair = TracePair(
+            dd_bdry, interior=kappa_minus, exterior=kappa_minus)
+        u_tpair = TracePair(
+            dd_bdry, interior=u_minus, exterior=u_minus)
+        normal = actx.thaw(dcoll.normal(dd_bdry))
+        return numerical_flux_func(kappa_tpair, u_tpair, normal)
+
+    def get_diffusion_flux(
+            self, dcoll, dd_bdry, kappa_minus, u_minus, grad_u_minus,
+            lengthscales_minus, *, penalty_amount=None,
+            numerical_flux_func=diffusion_facial_flux_harmonic):  # noqa: D102
+        actx = u_minus.array_context
+        normal = actx.thaw(dcoll.normal(dd_bdry))
+
+        external_flux = u_minus*0.0 + self.value
+
+        # flip the sign of the normal to indicate that positive values of
+        # flux are going in the cell
+        return np.dot(external_flux, -normal)
```

Comment:
I was following the notation that fluxes are positive into the system. To avoid confusion, I removed the signed and added a comment in the documentation.